clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesCredentials.java

```diff
@@ -28,21 +28,27 @@
 import java.util.*;
 
 public class KubernetesCredentials {
+  private final String name;
   private final KubernetesApiAdaptor apiAdaptor;
   private final List<String> namespaces;
   private final List<LinkedDockerRegistryConfiguration> dockerRegistries;
   private final HashMap<String, List<String>> imagePullSecrets;
   private final Logger LOG;
+  private final List<String> requiredGroupMembership;
 
-  public KubernetesCredentials(KubernetesApiAdaptor apiAdaptor,
+  public KubernetesCredentials(String name,
+                               KubernetesApiAdaptor apiAdaptor,
                                List<String> namespaces,
                                List<LinkedDockerRegistryConfiguration> dockerRegistries,
-                               AccountCredentialsRepository accountCredentialsRepository) {
+                               AccountCredentialsRepository accountCredentialsRepository,
+                               List<String> requiredGroupMembership) {
```

Comment:
Take a peek at https://github.com/spinnaker/clouddriver/blob/master/clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/AccountCredentials.java#L71

Notice that the NamedAccountCredentials objects all implement this interface, returning the real creds via that method.